app/controllers/api/v1/likes_controller.rb

```diff
@@ -66,9 +80,67 @@ def post_service
         @post_service ||= PostService.new(current_user)
       end
 
+      def comment_service
+        @comment_service ||= CommentService.new(current_user)
+      end
+
       def like_json(like)
         LikesPresenter.new(like).as_api_json
       end
+
+      def create_for_post
+        like_service.create_for_post(params[:post_id])
+
+        head :no_content
+      end
+
+      def create_for_comment
+        return unless comment_and_post_validate(params[:post_id], params[:comment_id])
+
+        like_service.create_for_comment(params[:comment_id])
+
+        head :no_content
+      end
+
+      def destroy_for_post
+        if like_service.unlike_post(params[:post_id])
+          head :no_content
+        else
+          render_error 410, "Like doesn’t exist"
+        end
+      end
+
+      def destroy_for_comment
+        return unless comment_and_post_validate(params[:post_id], params[:comment_id])
+
+        if like_service.unlike_comment(params[:comment_id])
+          head :no_content
+        else
+          render_error 410, "Like doesn’t exist"
+        end
+      end
+
+      def comment_and_post_validate(post_guid, comment_guid)
+        if !comment_exists(comment_guid) || !comment_is_for_post(post_guid, comment_guid)
+          render_error 404, "Comment not found for the given post"
+          false
+        else
+          true
+        end
+      end
+
+      def comment_exists(comment_guid)
+        comment = comment_service.find!(comment_guid)
+        comment ? true : false
+      rescue ActiveRecord::RecordNotFound
+        false
+      end
+
+      def comment_is_for_post(post_guid, comment_guid)
+        comments = comment_service.find_for_post(post_guid)
+        comment = comments.find {|comment| comment[:guid] == comment_guid }
+        comment ? true : false
+      end
```

Comment:
The `comment_exists` is kinda redundant, if the comment doesn't exist, it also doesn't exist "for the post", so only checking `comment_is_for_post` is enough.

Also in `comment_is_for_post` you can use `.exists?` directly on an SQL level, instead of looping through all comments with `find`.

Like this:
```suggestion
      def comment_and_post_validate(post_guid, comment_guid)
        if comment_is_for_post(post_guid, comment_guid)
          true
        else
          render_error 404, "Comment not found for the given post"
          false
        end
      end

      def comment_is_for_post(post_guid, comment_guid)
        comments = comment_service.find_for_post(post_guid)
        comments.exists?(guid: comment_guid)
      end
```

This way the whole validation is just a single sql-query, instead of 2 queries and looping through all comments in ruby.